luda-editor/new-client/src/render_psd_sprite.rs

```diff
@@ -0,0 +1,80 @@
+use super::*;
+use luda_rpc::SceneSprite;
+use psd_sprite::{decode_psd_sprite_from_bytes, PsdSprite, SpriteLoadedImages};
+use psd_sprite_render::RenderPsdSprite;
+use std::{
+    collections::HashMap,
+    error::Error,
+    sync::{Arc, RwLock},
+};
+
+lazy_static! {
+    static ref PSD_SPRITE_LOAD_STATE: PsdSpriteStorage = PsdSpriteStorage::new();
+}
+
+pub fn render_psd_sprite(ctx: &RenderCtx, scene_sprite: &SceneSprite, screen_wh: Wh<Px>) {
+    let Some(sprite_id) = &scene_sprite.sprite_id else {
+        return;
+    };
+
+    let Some(load_state) = PSD_SPRITE_LOAD_STATE.try_get(sprite_id) else {
+        let sprite_id = sprite_id.clone();
+        PSD_SPRITE_LOAD_STATE.set(sprite_id.clone(), PsdSpriteLoadState::Loading);
+        ctx.spawn(async move {
+            namui::log!("Loading PSD sprite: {}", sprite_id);
+            // TODO: Load PSD sprite from the server and cache.
+            let psd_bytes = namui::file::bundle::read("test.psd").await.unwrap();
+            let decode_result = decode_psd_sprite_from_bytes(&psd_bytes, "test.psd").await;
+
+            let load_state = decode_result.map_or_else(
+                |err| PsdSpriteLoadState::Error(err.into()),
+                |(psd_sprite, loaded_images)| PsdSpriteLoadState::Loaded {
+                    psd_sprite: Arc::new(psd_sprite),
+                    loaded_images: Arc::new(loaded_images),
+                },
+            );
+            PSD_SPRITE_LOAD_STATE.set(sprite_id.clone(), load_state);
+        });
+        return;
+    };
+
+    let PsdSpriteLoadState::Loaded {
+        psd_sprite,
+        loaded_images,
+    } = load_state.as_ref()
+    else {
+        return;
+    };
+
+    psd_sprite.render(ctx, scene_sprite, loaded_images.clone(), screen_wh);
+}
+
+enum PsdSpriteLoadState {
+    Loading,
+    Loaded {
+        psd_sprite: Arc<PsdSprite>,
+        loaded_images: Arc<SpriteLoadedImages>,
+    },
+    #[allow(unused)]
+    Error(Box<dyn Error + Send + Sync>),
+}
+
+struct PsdSpriteStorage {
```

Comment:
where do you use this storage??

Reply:
https://github.com/NamseEnt/namseent/pull/923/files#diff-8307c749c165fe3cef874cfec9499c563dd5bc4a669b4fb42d81d0aeeaf94d61R11-R13

Here

Reply:
then the name `PSD_SPRITE_LOAD_STATE` must be strange. it should be storage. I thought `PSD_SPRITE_LOAD_STATE` is the kind of enum value.